sound/soc/codecs/Kconfig

```diff
@@ -1686,6 +1686,8 @@ config SND_SOC_RT711_SDCA_SDW
 config SND_SOC_RT712_SDCA_SDW
 	tristate "Realtek RT712 SDCA Codec - SDW"
 	depends on SOUNDWIRE
+	depends on ACPI
+	select SND_SOC_SDCA
```

Comment:
yes, but in practice there are no users of this device outside of ACPI

we could revisit this and put the select somewhere else. Maybe the machine driver?

Reply:
I would handle this ACPI dependency in sound/soc/sdca to compile even if ACPI=n
The machine driver should select ACPI if it is destined to run on ACPI machine

Reply:
machine drivers depends on ACPI.

The problem is really the 'select' part, which overrides the dependencies. So the dependencies have to be moved to the highest level...

Reply:
you select SND_SOC_SDCA, SND_SOC_SDCA should not depend on ACPI, if I understand from your other comment, the sdca lib will compile even if no ACPI is selected?

Reply:
no, the sdca lib will not be compiled without ACPI, but the functions have a fallback with static inlines.

That would make e.g. the changes to drivers/soundwire/slave.c where we add those sdca functions completely harmless on non-ACPI platforms.

Reply:
I removed this RT712 change, for now it's not relevant. Only the MTL match needs SDCA in this PR, so I added a select SND_SOC_SDCA to avoid compilation issues reported by the kbuild bot.